### PR TITLE
release: Update image tags for v0.11.1

### PR DESCRIPTION
**Pull request by rafvasq**
#### Release checklist

**Create release tags (`v0.11.1`) in these repositories:**
 - [x] [`modelmesh`](https://github.com/kserve/modelmesh/releases)
 - [x] [`modelmesh-minio-examples`](https://github.com/kserve/modelmesh-minio-examples/releases)
 - [x] [`modelmesh-runtime-adapter`](https://github.com/kserve/modelmesh-runtime-adapter/releases)
 - [x] [`rest-proxy`](https://github.com/kserve/rest-proxy/releases)

**Verify image tags got pushed to [DockerHub](https://hub.docker.com/u/kserve):**
 - [x] [kserve/modelmesh](https://hub.docker.com/r/kserve/modelmesh/tags)
 - [x] [kserve/modelmesh-minio-examples](https://hub.docker.com/r/kserve/modelmesh-minio-examples/tags)
 - [x] [kserve/modelmesh-runtime-adapter](https://hub.docker.com/r/kserve/modelmesh-runtime-adapter/tags)
 - [x] [kserve/rest-proxy](https://hub.docker.com/r/kserve/rest-proxy/tags)

**Update versions in the following files:**
 - [x] `config/default/config-defaults.yaml`: edit the images tags for ...
     - [x] `kserve/modelmesh`
     - [x] `kserve/rest-proxy`
     - [x] `kserve/modelmesh-runtime-adapter` 
 - [x] `config/dependencies/quickstart.yaml`: change the `kserve/modelmesh-minio-examples` image tag to use the pinned version
 - [x] `config/manager/kustomization.yaml`: edit the `newTag`
 - [x] `docs/component-versions.md`: update the version and component versions
 - [x] `docs/install/install-script.md`: update the `RELEASE` variable in the `Installation` section to the new `release-*` branch name
 - [x] `docs/quickstart.md`: update the `RELEASE` variable in the _"Get the latest release"_ section to the new `release-*` branch name
 - [x] `scripts/setup_user_namespaces.sh`: change the `modelmesh_release` version

### CODE REVIEW

File: docs/component-versions.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Component versions
 
-The following table shows the component versions for the latest modelmesh-serving release (v0.11.0).
+The following table shows the component versions for the latest modelmesh-serving release (v0.11.1).
 | Component | Description | Upstream Revision |
 | - | - | - |
-| ModelMesh | Serves as a general-purpose model serving management/routing layer | [v0.11.0](https://github.com/kserve/modelmesh/tree/v0.11.0) |
-| ModelMesh Runtime Adapter | Contains the unified puller/runtime-adapter image | [v0.11.0](https://github.com/kserve/modelmesh-runtime-adapter/tree/v0.11.0) |
-| REST Proxy | Supports inference requests using KServe V2 REST Predict Protocol | [v0.11.0](https://github.com/kserve/rest-proxy/tree/v0.11.0) |
+| ModelMesh | Serves as a general-purpose model serving management/routing layer | [v0.11.1](https://github.com/kserve/modelmesh/tree/v0.11.0) |
```
Comment (ckadner) [inline docs/component-versions.md:6]: missed one :-)

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ on:
   push:
     branches:
       - main
-      - 'release-[0-9].[0-9]+'
+      - 'release-0.11.1'
```
Comment (ckadner) [inline .github/workflows/build.yml:8]: This is good to trigger the workflows here. Generally we don't want to create `release-x.x.x` branches for fixpatch releases so we keep this branch rule regex in the `release-0.11.1` branch.

File: controllers/modelmesh/modelmesh.go [hunk 3]
```diff
@@ -261,24 +261,24 @@ func (m *Deployment) addMMEnvVars(deployment *appsv1.Deployment) error {
 	}
 
 	if m.EnableAccessLogging {
-		// See https://github.com/kserve/modelmesh/blob/v0.11.0/src/main/java/com/ibm/watson/modelmesh/ModelMeshEnvVars.java#L55
+		// See https://github.com/kserve/modelmesh/blob/v0.11.1/src/main/java/com/ibm/watson/modelmesh/ModelMeshEnvVars.java#L55
```
Comment (ckadner) [inline controllers/modelmesh/modelmesh.go:264]: did you check if those line numbers stayed the same after the changes that went into the `modelmesh` repo?

**Review by ckadner — CHANGES_REQUESTED** [review]
Thanks @rafvasq -- I have just a few comments

File: controllers/modelmesh/modelmesh.go (hunk 3 above)
Comment (rafvasq) [reply controllers/modelmesh/modelmesh.go:264]: Yup, they're the same.

File: docs/component-versions.md (hunk 1 above)
Comment (rafvasq) [reply docs/component-versions.md]: thanks, nice catch!

**Review by ckadner — APPROVED** [review]
/lgtm

**Pull request merged**